src/vaadin-date-picker-mixin.html

```diff
@@ -860,7 +893,8 @@
       // Note (Yuriy): Workaround `this._inputValue` is used in order
       // to avoid breaking change on custom `checkValidity`.
       // Can be removed with next major.
-      const isValid = this.checkValidity(this._inputValue);
+      // Note : this._inputElementValue is used instead of this._inputValue
```

Comment:
Let's clarify the comment a bit.

```
Note (Yuriy): `this._inputElementValue` is passed to `checkValidity()` for backward compatibility in case someone is relying on it in a custom `checkValidity()` implementation. The workaround can be removed with the next major.
```